cmd/gardenlet/app/migration.go

```diff
@@ -6,10 +6,150 @@ package app
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardeneraccess "github.com/gardener/gardener/pkg/component/gardener/access"
+	shootsystem "github.com/gardener/gardener/pkg/component/shoot/system"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
 )
 
-func (g *garden) runMigrations(_ context.Context, _ logr.Logger) error {
+func (g *garden) runMigrations(ctx context.Context, log logr.Logger) error {
+	log.Info("Migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig")
+	if err := migrateAdminViewerKubeconfigClusterRoleBindings(ctx, log, g.mgr.GetClient()); err != nil {
+		return fmt.Errorf("failed migrating ClusterRoleBindings for shoot/adminkubeconfig and shoot/viewerkubeconfig: %w", err)
+	}
+
 	return nil
 }
+
+// migrateAdminViewerKubeconfigClusterRoleBindings moves the ClusterRoleBindings granting access to the
+// shoot/adminkubeconfig and shoot/viewerkubeconfig subresources from the shoot-core-system managed resource to the
+// shoot-core-gardeneraccess managed resource.
+// TODO(vpnachev): Remove this after v1.132.0 has been released.
+func migrateAdminViewerKubeconfigClusterRoleBindings(ctx context.Context, log logr.Logger, seedClient client.Client) error {
+	namespaceList := &corev1.NamespaceList{}
+	if err := seedClient.List(ctx, namespaceList, client.MatchingLabels(map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot})); err != nil {
+		return fmt.Errorf("failed listing namespaces: %w", err)
+	}
+
+	var (
+		tasks []flow.TaskFn
+		crbs  = []string{v1beta1constants.ShootProjectAdminsGroupName, v1beta1constants.ShootProjectViewersGroupName, v1beta1constants.ShootSystemAdminsGroupName, v1beta1constants.ShootSystemViewersGroupName}
+	)
+
+	for _, namespace := range namespaceList.Items {
+		if namespace.DeletionTimestamp != nil || namespace.Status.Phase == corev1.NamespaceTerminating {
+			continue
+		}
+
+		tasks = append(tasks, func(ctx context.Context) error {
+			var (
+				shootSystemKey             = client.ObjectKey{Namespace: namespace.Name, Name: shootsystem.ManagedResourceName}
+				shootSystemManagedResource = &resourcesv1alpha1.ManagedResource{}
+
+				gardenerAccessKey             = client.ObjectKey{Namespace: namespace.Name, Name: gardeneraccess.ManagedResourceName}
+				gardenerAccessManagedResource = &resourcesv1alpha1.ManagedResource{}
+			)
+
+			// Get the shoot-core-system managed resource and check if it is already migrated.
+			if err := seedClient.Get(ctx, shootSystemKey, shootSystemManagedResource); err != nil {
+				if apierrors.IsNotFound(err) {
+					log.Info("Managed resource not found, skipping migration", "managedResource", shootSystemKey)
+					return nil
+				}
+				return fmt.Errorf("failed to get ManagedResource %q: %w", shootSystemKey, err)
+			}
+
+			if shootSystemManagedResource.DeletionTimestamp != nil {
+				return nil
```

Comment:
We could add a log entry for this early exit:
```suggestion
				log.Info("Managed resource is in deletion, skipping migration", "managedResource", shootSystemKey)
				return nil
```